pyproject.toml

```diff
@@ -43,7 +43,7 @@ test = [
     'pytest',
     'pytest-cov',
     'pytest-xdist',
-    'ruff',
+    'ruff>=0.9',
```

Comment:
ruff 0.8 will undo many of the formatting changes in ruff 0.9, let's set this as the new minimum version required.